File: SchedulerSite/QuartzScheduler/src/main/java/com/dfssi/dataplatform/quartz/util/MathUtil.java
package com.dfssi.dataplatform.quartz.util;

import java.math.BigDecimal;
import java.math.RoundingMode;

/**
 * Description:
 *
 * @author LiXiaoCong
 * @version 2018/5/17 8:48
 */
public class MathUtil {
    private MathUtil(){}

    public static double rounding(double value, int p) {
        BigDecimal b = new BigDecimal(value);
        return b.setScale(p, RoundingMode.HALF_UP).doubleValue();
    }
}

File: DataMiningSite/DataMiningAnalysis/src/main/scala/com/dfssi/dataplatform/analysis/util/MathUtil.scala
package com.dfssi.dataplatform.analysis.util

import java.math.RoundingMode


/**
  * Description:
  *
  * @author LiXiaoCong
  * @version 2018/5/16 14:08 
  */
object MathUtil {

    def rounding(value: Double, p: Int): Double ={
        val b = new java.math.BigDecimal(value)
        b.setScale(p, RoundingMode.HALF_UP).doubleValue
    }

}
